config/300-samplesource.yaml

```diff
@@ -0,0 +1,137 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  labels:
+    samples.knative.dev/release: devel
+    eventing.knative.dev/source: "true"
+    knative.dev/crd-install: "true"
+  annotations:
+    registry.knative.dev/eventTypes: |
+      [
+        { "type": "dev.knative.sample.source" }
+      ]
+  name: samplesources.samples.knative.dev
+spec:
+  group: samples.knative.dev
+  names:
+    categories:
+    - all
+    - knative
+    - eventing
+    - sources
+    kind: SampleSource
+    plural: samplesources
+  scope: Namespaced
+  subresources:
+    status: {}
+  validation:
+    openAPIV3Schema:
+      properties:
+        apiVersion:
+          type: string
+        kind:
+          type: string
+        metadata:
+          type: object
+        spec:
+          properties:
+            serviceAccountName:
+              type: string
+              description: >
+                ServiceAccountName holds the name of the Kubernetes service account
+                as which the sample source should run. If unspecified
+                this will default to the "default" service account for the namespace
+                in which the SampleSource exists
+            interval:
+              type: string
+              minLength: 2
+              description:
+                Interval between events, number and unit. For example "5s", "100ms"
+            sink:
+              anyOf:
+              - type: object
+                description: "The destination that should receive events"
+                properties:
+                  ref:
+                    type: object
+                    description: "The reference to a Kubernetes object from which to retrieve the target URI"
+                    required:
+                    - apiVersion
+                    - kind
+                    - name
+                    properties:
+                      apiVersion:
+                        type: string
+                        minLength: 1
+                      kind:
+                        type: string
+                        minLength: 1
+                      name:
+                        type: string
+                        minLength: 1
+                  uri:
+                    type: string
+                    description: "The target URI. If ref is provided, this must be a relative URI reference"
+              - type: object
```

Comment:
maybe not support this since this is new and might be better to not start with the deprecated fields?

Reply:
￼fixed in e7eabf7